lib/mpl_toolkits/axisartist/axislines.py

```diff
@@ -95,38 +95,51 @@ def update_lim(self, axes):
         delta2 = _api.deprecated("3.6")(
             property(lambda self: 0.00001, lambda self, value: None))
 
+        def _to_xy(self, values, const):
+            """
+            Create a (len(values), 2)-shape array representing (x, y) pairs.
+
+            *values* go into the coordinate determined by ``self.nth_coord``.
+            The other coordinate is filled with the constant *const*.
+
+            Example::
+
+                >>> self.nth_coord = 0
+                >>> self._to_xy([1, 2, 3], const=0)
+                array([[1, 0],
+                       [2, 0],
+                       [3, 0]])
+            """
+            if self.nth_coord == 0:
+                return np.stack(np.broadcast_arrays(values, const), axis=-1)
+            elif self.nth_coord == 1:
+                return np.stack(np.broadcast_arrays(const, values), axis=-1)
+            else:
+                raise ValueError("Unexpected nth_coord")
+
     class Fixed(_Base):
         """Helper class for a fixed (in the axes coordinate) axis."""
 
+        # deprecated with passthru_pt
         _default_passthru_pt = dict(left=(0, 0),
                                     right=(1, 0),
                                     bottom=(0, 0),
                                     top=(0, 1))
+        passthru_pt = _api.deprecated("3.7")(property(
+            lambda self: self._default_passthru_pt[self._loc]))
 
         def __init__(self, loc, nth_coord=None):
             """``nth_coord = 0``: x-axis; ``nth_coord = 1``: y-axis."""
             _api.check_in_list(["left", "right", "bottom", "top"], loc=loc)
             self._loc = loc
-
-            if nth_coord is None:
-                if loc in ["left", "right"]:
-                    nth_coord = 1
-                else:  # "bottom", "top"
-                    nth_coord = 0
-
-            self.nth_coord = nth_coord
-
+            self._pos = _api.check_getitem(
+                {"bottom": 0, "top": 1, "left": 0, "right": 1}, loc=loc)
```

Comment:
```suggestion
            self._pos = {"bottom": 0, "top": 1, "left": 0, "right": 1}[loc]
```

We check 2 lines above that the string is in the list of 4 acceptable inputs so this should always work.

Reply:
Indeed, fixed.